public/scripts/main.js

```diff
@@ -38,23 +38,40 @@ const keyToId = {
 // this tracks when we started asking for the current key command
 let questionStartMS = 0;
 
-$(document).ready(function() {
+$(document).ready(function () {
   //$("#retryButton").toggleClass("on");
   //alert($('li[data-keycode="test"]').attr('id'));
-   fetch('scripts/shortcuts.json')
-  .then(response => response.json())
-  .then(data => {
-    allData=data
-    if(localStorage.getItem("questionNo")==null)
-    {
-      localStorage.setItem("questionNo", "1");
-      localStorage.setItem("totalCount", Object.keys(allData).length);
-    }
-     readText()
-     updateTimingDisplay()
+  fetch('scripts/shortcuts.json')
+    .then(response => response.json()).then(data => {
+      allData = data
+      return getUserProgress()
+    }).then(response => {
+      if (!response.ok) {
+        const message = `An error has occured: ${response.status}`;
+        console.error(message)
+        return 1
```

Comment:
Not sure if you missed the other PR comment but the `return 1` won't work because the next step is expecting a `userProgress` object